central/complianceoperator/v2/report/manager/watcher/scanwatcher_test.go

```diff
@@ -0,0 +1,324 @@
+	case <-time.After(100 * time.Millisecond):
+		t.Error("timeout waiting for the watcher to stop")
+	}
+	// We should have a result in the queue with an error
+	require.Equal(t, 1, resultQueue.Len())
+	result := resultQueue.Pull()
+	assert.Error(t, result.Error)
+}
+
+func TestGetIDFromScan(t *testing.T) {
+	scan := &storage.ComplianceOperatorScanV2{}
+	_, err := GetWatcherIDFromScan(scan)
+	assert.Error(t, err)
+	scan.ClusterId = "cluster-1"
+	_, err = GetWatcherIDFromScan(scan)
+	assert.Error(t, err)
+	scan.Id = "scan-1"
+	id, err := GetWatcherIDFromScan(scan)
+	assert.NoError(t, err)
+	assert.Equal(t, "", id)
+	scan.LastStartedTime = protocompat.TimestampNow()
+	id, err = GetWatcherIDFromScan(scan)
+	assert.NoError(t, err)
+	assert.Equal(t, fmt.Sprintf("%s:%s:%s", scan.ClusterId, scan.Id, scan.LastStartedTime), id)
+}
+
+func TestGetIDFromResult(t *testing.T) {
+	timeNow := protocompat.TimestampNow()
+	ctrl := gomock.NewController(t)
+	ds := mocks.NewMockDataStore(ctrl)
+	// Error querying the Scan DataStore
+	ds.EXPECT().SearchScans(gomock.Any(), gomock.Any()).Times(1).
+		DoAndReturn(func(_, _ any) ([]*storage.ComplianceOperatorScanV2, error) {
+			return nil, errors.New("db error")
+		})
+	result := &storage.ComplianceOperatorCheckResultV2{}
+	_, err := GetWatcherIDFromCheckResult(testDBAccess, result, ds)
+	assert.Error(t, err)
+	// No Scan retrieved
+	ds.EXPECT().SearchScans(gomock.Any(), gomock.Any()).Times(1).
+		DoAndReturn(func(_, _ any) ([]*storage.ComplianceOperatorScanV2, error) {
+			return nil, nil
+		})
+	_, err = GetWatcherIDFromCheckResult(testDBAccess, result, ds)
+	assert.Error(t, err)
+	// Scan retrieved successfully
+	ds.EXPECT().SearchScans(gomock.Any(), gomock.Any()).Times(4).
+		DoAndReturn(func(_, _ any) ([]*storage.ComplianceOperatorScanV2, error) {
+			return []*storage.ComplianceOperatorScanV2{
+				{
+					ClusterId:       "cluster-1",
+					Id:              "scan-1",
+					LastStartedTime: timeNow,
+				},
+			}, nil
+		})
+	// Empty annotation
+	_, err = GetWatcherIDFromCheckResult(testDBAccess, result, ds)
+	assert.Error(t, err)
+	// Invalid format in the annotation
+	result.Annotations = map[string]string{
+		lastScannedAnnotationKey: protocompat.TimestampNow().String(),
+	}
+	// The timestamp does not coincide with the Scan start time
+	_, err = GetWatcherIDFromCheckResult(testDBAccess, result, ds)
+	assert.Error(t, err)
+	result.Annotations = map[string]string{
+		lastScannedAnnotationKey: protocompat.TimestampNow().AsTime().Format(time.RFC3339Nano),
+	}
+	// Successful execution
+	_, err = GetWatcherIDFromCheckResult(testDBAccess, result, ds)
+	assert.Error(t, err)
+	result.Annotations = map[string]string{
+		lastScannedAnnotationKey: timeNow.AsTime().Format(time.RFC3339Nano),
+	}
+	id, err := GetWatcherIDFromCheckResult(testDBAccess, result, ds)
+	assert.NoError(t, err)
+	assert.Equal(t, fmt.Sprintf("cluster-1:scan-1:%s", timeNow.String()), id)
+}
+
+func TestIsComplianceOperatorHealthy(t *testing.T) {
```

Comment:
Referring to my other comment about this function - it is tested, so I have nothing against keeping it in this PR.